diff --git a/languages/cpp/ast/ast_cpp.ml b/languages/cpp/ast/ast_cpp.ml
index 304b33f72..cde2cd615 100644
--- a/languages/cpp/ast/ast_cpp.ml
+++ b/languages/cpp/ast/ast_cpp.ml
@@ -1201,8 +1201,7 @@ let (ii_of_name : name -> tok) =
     | IdConverter (tok, _ft) -> tok
     | IdDestructor (tok, (_s, _ii)) -> tok
     | IdTemplated (x, _args) -> ident_or_op x
-    (* TODO? *)
-    | IdDeref _ -> failwith "ii_of_name: IdDeref"
+    | IdDeref (tok, _x) -> tok
   in
   ident_or_op id
 
diff --git a/languages/cpp/tree-sitter/Parse_cpp_tree_sitter.ml b/languages/cpp/tree-sitter/Parse_cpp_tree_sitter.ml
index fb8fdb190..1ad46b350 100644
--- a/languages/cpp/tree-sitter/Parse_cpp_tree_sitter.ml
+++ b/languages/cpp/tree-sitter/Parse_cpp_tree_sitter.ml
@@ -5512,9 +5512,9 @@ and map_type_declarator_to_expr (env : env) (x : CST.type_declarator) : expr =
       let v5 = map_type_declarator_to_expr env v5 in
       Unary ((DeRef, v2), v5)
   | `Func_type_decl (v1, v2) ->
-      let _v1 = map_type_declarator_to_expr env v1 in
+      let v1 = map_type_declarator_to_expr env v1 in
       let _v2 = map_parameter_list env v2 in
-      failwith "unallowed declarator as RHS to scope resolution"
+      v1
   | `Array_type_decl (v1, v2, v3, v4, v5) -> (
       let v1 = map_type_declarator_to_expr env v1 in
       let v2 =
